internal/api/handlers/home_domains.go

```diff
@@ -0,0 +1,175 @@
+package handlers
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"maps"
+	"net/http"
+	"slices"
+
+	"github.com/stellar/freighter-backend-v2/internal/api/httperror"
+	response "github.com/stellar/freighter-backend-v2/internal/api/httpresponse"
+	"github.com/stellar/freighter-backend-v2/internal/types"
+	"github.com/stellar/go/xdr"
+)
+
+
+type HomeDomainsHandler struct {
+	RpcService types.RPCService
+
+}
+
+type HomeDomainMap map[string]string
+
+
+func NewHomeDomainsHandler(rpc types.RPCService) *HomeDomainsHandler {
+	return &HomeDomainsHandler{
+		RpcService: rpc,
+	}
+}
+
+type PublicKeyError struct {
+	PublicKey    string `json:"public_key"`
+	ErrorMessage string `json:"error_message"`
+}
+
+type HomeDomainsError struct {
+	ErrorMessage      string       `json:"error_message"`
+	Error_keys        []PublicKeyError `json:"error_keys,omitempty"`
+}
+
+type HomeDomainsResponse struct {
+	HomeDomains map[string]string `json:"home_domains"` // Removed omitempty
+	Error       HomeDomainsError `json:"error,omitempty"`
+}
+
+type LedgerKeyAccountKeys struct {
+	LedgerKeys []string `json:"ledger_keys"`
+	HomeDomainMap HomeDomainMap `json:"home_domain_map"`
+}
+
+func getLedgerKeyAccountKeys(publicKeys []string) (LedgerKeyAccountKeys, HomeDomainsError) {
```

Comment:
I tried to follow the coding conventions used in the `/collectibles` endpoint here